Deprecation warning about implicit copy constructor

> D:/builds/turtle/include\turtle/detail/expectation_template.hpp:158:9: warning: definition of implicit copy constructor for 'expectation<TYPE_T ()>' is deprecated because it has a user-declared destructor [-Wdeprecated-copy-dtor]
>         ~expectation()
>         ^
> D:/builds/linuxhost/linux64/10.2.0/x86_64/lib/gcc/x86_64-i3linuxhost-linux-gnu/10.2.0/../../../../x86_64-i3linuxhost-linux-gnu/include/c++/10.2.0\ext/new_allocator.h:150:23: note: in implicit copy constructor for 'mock::detail::expectation<TYPE_T ()>' first required here
>         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
>                              ^
> D:/builds/linuxhost/linux64/10.2.0/x86_64/lib/gcc/x86_64-i3linuxhost-linux-gnu/10.2.0/../../../../x86_64-i3linuxhost-linux-gnu/include/c++/10.2.0\bits/alloc_traits.h:512:8: note: in instantiation of function template specialization '__gnu_cxx::new_allocator<std::_List_node<mock::detail::expectation<TYPE_T ()> > >::construct<mock::detail::expectation<TYPE_T ()>, mock::detail::expectation<TYPE_T ()> >' requested here
>           __a.construct(__p, std::forward<_Args>(__args)...);
>               ^
> D:/builds/linuxhost/linux64/10.2.0/x86_64/lib/gcc/x86_64-i3linuxhost-linux-gnu/10.2.0/../../../../x86_64-i3linuxhost-linux-gnu/include/c++/10.2.0\bits/stl_list.h:637:24: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<std::_List_node<mock::detail::expectation<TYPE_T ()> > > >::construct<mock::detail::expectation<TYPE_T ()>, mock::detail::expectation<TYPE_T ()> >' requested here
>           _Node_alloc_traits::construct(__alloc, __p->_M_valptr(),
>                               ^
> D:/builds/linuxhost/linux64/10.2.0/x86_64/lib/gcc/x86_64-i3linuxhost-linux-gnu/10.2.0/../../../../x86_64-i3linuxhost-linux-gnu/include/c++/10.2.0\bits/stl_list.h:1911:18: note: in instantiation of function template specialization 'std::__cxx11::list<mock::detail::expectation<TYPE_T ()>, std::allocator<mock::detail::expectation<TYPE_T ()> > >::_M_create_node<mock::detail::expectation<TYPE_T ()> >' requested here
>          _Node* __tmp = _M_create_node(std::forward<_Args>(__args)...);
>                         ^
> D:/builds/linuxhost/linux64/10.2.0/x86_64/lib/gcc/x86_64-i3linuxhost-linux-gnu/10.2.0/../../../../x86_64-i3linuxhost-linux-gnu/include/c++/10.2.0\bits/stl_list.h:1217:15: note: in instantiation of function template specialization 'std::__cxx11::list<mock::detail::expectation<TYPE_T ()>, std::allocator<mock::detail::expectation<TYPE_T ()> > >::_M_insert<mock::detail::expectation<TYPE_T ()> >' requested here
>       { this->_M_insert(end(), std::move(__x)); }
>               ^
> D:/builds/turtle/include\turtle/detail/function_impl_template.hpp:217:27: note: in instantiation of member function 'std::__cxx11::list<mock::detail::expectation<TYPE_T ()>, std::allocator<mock::detail::expectation<TYPE_T ()> > >::push_back' requested here
>             expectations_.push_back( expectation_type( file, line ) );
>                           ^
> D:/builds/turtle/include\turtle/detail/function_template.hpp:67:27: note: in instantiation of member function 'mock::detail::function_impl<TYPE_T ()>::expect' requested here
>             return impl_->expect( file, line );
>                           ^
> test_connection_management.cpp:437:17: note: in instantiation of member function 'mock::detail::function<TYPE_T ()>::expect' requested here
>                 MOCK_EXPECT(pTran->on_timeout).never();
>                 ^
> D:/builds/turtle/include\turtle/mock.hpp:249:39: note: expanded from macro 'MOCK_EXPECT'
> #define MOCK_EXPECT(t) MOCK_HELPER(t).expect( __FILE__, __LINE__ )
>                                       ^
>

--- a/include/turtle/detail/expectation_template.hpp
+++ b/include/turtle/detail/expectation_template.hpp
@@ -155,6 +155,11 @@ namespace detail
             , line_( line )
         {}
 
+        expectation(expectation &&) = default;
+        expectation(expectation const&) = default;
+        expectation& operator=(expectation &&) = default;
+        expectation& operator=(expectation const&) = default;
+
         ~expectation()
         {
             for( sequences_cit it = sequences_.begin();

